### PR TITLE
setting up new term "MetaboliteIntensity"

### PR DESCRIPTION
**Pull request by josephpowi**
## Description

This PR allows to handle the metaboliteIntensity type in pp. Now we can search, edit and filter by the term type. It can be used as a reference of what needs to be done to support a new term type. To test it you would need to set up the dataset redomics.mm10.ts. This dataset also uses a new genome, in server config you will need to add this.

{
	"name": "mm10",
	"file": "./genome/mm10.ts",
	"species": "mouse",
	"datasets": [
	{ "name": "redomics", "jsfile": "./dataset/redomics.mm10.ts"}]}
}

and you will need to download the genome. This can be done running from pp utils/mm10.sh. Please check.

Initial changes:

1. dev/sjpp/proteinpaint/server/shared/types/terms/metaboliteIntensity.ts
	MetaboliteIntensityQ
	MetaboliteIntensityTermWrapper
	MetaboliteIntensityTermSettingInstance

2. server/shared/types/dataset.ts
	MetaboliteIntensityQueryNative
	MetaboliteIntensityQuery 

3. #shared/common.js 
	dtmetaboliteintensity

	dt2label
		[dtmetaboliteintensity]: 'Metabolite Intensity'
	TermTypes
		METABOLITE_INTENSITY: 'metaboliteIntensity'
	TermTypeGroups
		METABOLITE_INTENSITY: 'Metabolite Intensity'

4. server/routes/termdb.cluster.ts 
	add: MetaboliteIntensityQuery, MetaboliteIntensityQueryNative
		 dtmetaboliteintensity
	validate_query_metaboliteIntensity
	validateMetaboliteIntensityNative

5. server/src/utils.js 
	validate_txtfile
	get_header_txt
	get_lines_txtfile

6. termdb.cluster.ts
	validateMetaboliteIntensityNative
	validate_query_metaboliteIntensity

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Comment by josephpowi** [comment]
I resolved the txt file header reading problem

**Comment by airenzp** [comment]
We can now search by Metabolite Intensity from the dictionary but we need to solve some issues searching the metabolite in the metabolomics.get. I am marking this PR as a draft until this is solved.

File: server/routes/termdb.cluster.ts [hunk 1]
```diff
@@ -251,3 +256,86 @@ async function validateNative(q: GeneExpressionQueryNative, ds: any, genome: any
 		return { gene2sample2value, byTermId, bySampleId }
 	}
 }
+
+export async function validate_query_metaboliteIntensity(ds: any, genome: any) {
+	const q: MetaboliteIntensityQuery = ds.queries.metabolomics
```
Comment (xzhou82) [inline server/routes/termdb.cluster.ts:261]: please change to `ds.queries.metaboliteIntensity{}`

**Review by airenzp — DISMISSED** [review]

**Review by xzhou82 — APPROVED** [review]
prototype looks good!

**Pull request merged**